server/admin_management/api/app/routers/router_application_admin.py

```diff
@@ -0,0 +1,168 @@
+from sqlalchemy.orm import Session
+from typing import List
+
+
+from api.app.models import model as models
+from api.app.routers.router_guards import (
+    get_current_requester,
+    authorize_by_fam_admin,
+    enforce_self_grant_guard,
+)
+from api.app import database, jwt_validation, schemas
+from api.app.schemas import Requester
+from api.app.services.application_admin_service import ApplicationAdminService
+from api.app.services.user_service import UserService
+from api.app.services.application_service import ApplicationService
+from api.app.utils.audit_util import AuditEventLog, AuditEventOutcome, AuditEventType
+
+LOGGER = logging.getLogger(__name__)
+
+router = APIRouter()
+
+
+@router.post(
+    "",
+    response_model=schemas.FamAppAdminGet,
+    dependencies=[Depends(authorize_by_fam_admin), Depends(enforce_self_grant_guard)],
+)
+def create_application_admin(
+    application_admin_request: schemas.FamAppAdminCreate,
+    request: Request,
+    db: Session = Depends(database.get_db),
+    token_claims: dict = Depends(jwt_validation.authorize),
+    requester: Requester = Depends(get_current_requester),
+):
+
+    LOGGER.debug(
+        f"Executing 'create_user_role_assignment' "
+        f"with request: {application_admin_request}, requestor: {token_claims}"
+    )
+
+    audit_event_log = AuditEventLog(
+        request=request,
+        event_type=AuditEventType.CREATE_APPLICATION_ADMIN_ACCESS,
+        event_outcome=AuditEventOutcome.SUCCESS,
+    )
+
+    try:
+        application_admin_service = ApplicationAdminService(db)
+        application_service = ApplicationService(db)
+        user_service = UserService(db)
+
+        audit_event_log.requesting_user: models.FamUser = (
+            user_service.get_user_by_cognito_user_id(requester.cognito_user_id)
+        )
+        audit_event_log.application: models.FamApplication = (
+            application_service.get_application(
+                application_admin_request.application_id
+            )
+        )
+        audit_event_log.target_user: models.FamUser = (
+            user_service.get_user_by_domain_and_name(
+                application_admin_request.user_type_code,
+                application_admin_request.user_name,
+            )
+        )
+
+        return application_admin_service.create_application_admin(
+            application_admin_request, requester.cognito_user_id
+        )
+
+    except Exception as e:
+        audit_event_log.event_outcome = AuditEventOutcome.FAIL
+        audit_event_log.exception = e
+        raise e
+
+    finally:
+        if audit_event_log.target_user is None:
+            audit_event_log.target_user = models.FamUser(
+                user_type_code=application_admin_request.user_type_code,
+                user_name=application_admin_request.user_name,
+                user_guid="unknown",
+                cognito_user_id="unknown",
+            )
+
+        audit_event_log.log_event()
```

Comment:
I think this is because when we query the target_user it could be None, but when we call the creat_application_admin service, the new user will be created if not exists. So in the "finally" section, we'll add the new user as target_user

Reply:
Ah~ I see, I missed that point that user can be created and persisted later on.
Then it looks like, even with exception happens, the target_user is still logged, same as original api, hmm...